src/recommendation/xml.ts

```diff
@@ -0,0 +1,31 @@
+import * as vscode from "vscode";
+import { sendInfo } from "vscode-extension-telemetry-wrapper";
+
+import { validateAndRecommendExtension } from "../utils";
+import { instrumentCommand } from "../command";
+
+const KEY_XML_RECOMMENDATION_LAST_SHOW_TIME = "xmlRecommendationLastShowTime";
+
+export function initialize (context: vscode.ExtensionContext) {
+  const recommendationHandler = instrumentCommand(context, "recommendExtension", extensionRecommendationHandler);
+  context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(e => {
+    if (e.uri.path.toLowerCase().indexOf("pom.xml") !== -1) {
```

Comment:
=> `if (e.fileName === "pom.xml")`

Reply:
`e.filename.toLowerCase()` for robustness? I've seen `POM.XML` in some projects.

Reply:
e.fileName is the full path. === sign cannot be used directly to check whether the current file is pom.xml